src/type-translator.ts

```diff
@@ -532,7 +531,8 @@ export class TypeTranslator {
       return `function(new: (${constructedType})${paramsStr}): ?`;
     }
 
-    for (const field of toArray(type.symbol.members.keys())) {
+    // tslint:disable-next-line:no-any
+    for (const field of (type.symbol.members.keys() as any)) {
```

Comment:
Why is this needed?  Isn't members already a Map?

Reply:
It's a ts.Map, which wants to be a real Map, but isn't for some reason.

Reply:
It looks like it's just named Map, but it's not actually compatible with ES6.  I would guess this code doesn't work as written.

file: 'file:///usr/local/google/home/evanm/projects/tsickle/src/type-translator.ts'
severity: 'Error'
message: 'Type 'Iterator<string>' is not an array type or a string type or does not have a '[Symbol.iterator]()' method that returns an iterator.'
at: '535,21'
source: 'ts'

Reply:
The comments on the type say it's an ES6 map.  I'm pretty sure that the object is a proper map, they just produced a bad .d.ts for some reason.  All the tests pass, which wouldn't be the case if this threw an error instead of doing the right thing.

Reply:
Can you add a comment explaining this?

Reply:
Done.